### PR TITLE
Add tensorrt test workflow

### PR DESCRIPTION
**Pull request by lanluo-nvidia**
# Description

Added the tensorrt test workflow:
example run: 
https://github.com/pytorch/TensorRT/actions/runs/11675576873/job/32510381587?pr=3266

## Type of change

Please delete options that are not relevant and/or add your own.

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- This change requires a documentation update

# Checklist:

- [ ] My code follows the style guidelines of this project (You can use the linters)
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas and hacks
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests to verify my fix or my feature
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have added the relevant labels to my PR in so that relevant reviewers are notified

### CODE REVIEW

File: .github/workflows/build-tensorrt-linux.yml [hunk 1]
```diff
@@ -0,0 +1,222 @@
+name: Build Torch-TensorRT wheel on Linux with specified tensorRT version
```
Comment (narendasan) [inline .github/workflows/build-tensorrt-linux.yml:1]: Can we name this something a bit shorter and straight forward such as `Build Torch-TensorRT w/ Future TensorRT Versions`

File: .github/workflows/build-test-tensorrt-linux.yml [hunk 2]
```diff
@@ -0,0 +1,136 @@
+name: Build and test Torch-TensorRT on Linux with specified tensorRT version
```
Comment (narendasan) [inline .github/workflows/build-test-tensorrt-linux.yml:1]: Similar here

File: packaging/pre_build_script.sh [hunk 3]
```diff
@@ -21,6 +21,15 @@ pip install --force-reinstall --pre ${TORCH_TORCHVISION} --index-url ${INDEX_URL
 export TORCH_BUILD_NUMBER=$(python -c "import torch, urllib.parse as ul; print(ul.quote_plus(torch.__version__))")
 export TORCH_INSTALL_PATH=$(python -c "import torch, os; print(os.path.dirname(torch.__file__))")
 
+if [[ ${TENSORRT_VERSION} != "" ]]; then
+  # this is the upgraded TensorRT version, replace current tensorrt version to the upgrade tensorRT version in the pyproject.toml
+  current_version=$(cat dev_dep_versions.yml | grep __tensorrt_version__ | sed 's/__tensorrt_version__: //g' | sed 's/"//g')
+  sed -i -e "s/tensorrt-cu12==${current_version}/tensorrt-cu12==${TENSORRT_VERSION}/g" \
```
Comment (narendasan) [inline packaging/pre_build_script.sh:27]: Is CUDA 11.8 still in the support matrix?

File: packaging/pre_build_script.sh [hunk 4]
```diff
@@ -21,6 +21,15 @@ pip install --force-reinstall --pre ${TORCH_TORCHVISION} --index-url ${INDEX_URL
...
+         -e "s/tensorrt-cu12-bindings==${current_version}/tensorrt-cu12-bindings==${TENSORRT_VERSION}/g" \
+         -e "s/tensorrt-cu12-libs==${current_version}/tensorrt-cu12-libs==${TENSORRT_VERSION}/g" \
+          pyproject.toml
+fi
+
 if [[ "${CU_VERSION::4}" < "cu12" ]]; then
   # replace dependencies from tensorrt-cu12-bindings/libs to tensorrt-cu11-bindings/libs
   sed -i -e "s/tensorrt-cu12==/tensorrt-${CU_VERSION::4}==/g" \
```
Comment (narendasan) [inline packaging/pre_build_script.sh:35]: Or does it get handled here?

**Review by narendasan — COMMENTED** [review]
Mostly seems fine

File: packaging/pre_build_script.sh (hunk 3 above)
Comment (lanluo-nvidia) [reply packaging/pre_build_script.sh:27]: yes, cuda 11.8 still in the support matrix in tensorRT10.6.0

File: packaging/pre_build_script.sh (hunk 4 above)
Comment (lanluo-nvidia) [reply packaging/pre_build_script.sh:35]: yes, the cuda 11.8 is handled here.

**Comment by lanluo-nvidia** [comment]
> Seems like 12.1 got dropped is that us or PyTorch?

@narendasan 
yes, cuda 12.1 is deprecated by pytorch, it was merged in just a few hours ago.
https://github.com/pytorch/test-infra/pull/5904

Very wierd is that cuda11.8 is still supported, they only deprecated cuda12.1
